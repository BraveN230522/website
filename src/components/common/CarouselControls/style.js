import styled from '@emotion/styled';

import carouselBack from '@public/svg/carousel-back.svg';
import carouselNext from '@public/svg/carousel-next.svg';

const ControlStyles = styled.div`
  display: flex;
  ${({ isMobile }) =>
    isMobile
      ? `
  align-items: center;
  justify-content: center;
  width: 50%;
  max-width: calc(100vw - 150px);`
      : 'padding-bottom: 30px;'}
`;

const ControlItem = styled.span`
  width: ${({ active }) => (active ? 31 : 13)}px;
  height: 5px;
  display: block;
  border-radius: 4px;
  background-color: ${({ theme, active, controlColor, activeColor }) =>
    activeColor && active ? theme.colors[activeColor] : theme.colors[controlColor]};
  margin-left: 4px;
  cursor: pointer;
  transition: all ease-in 0.2s;

  &:hover {
    width: 31px;
  }
`;

const ControlBackNext = styled.div`
  height: 60px;
  width: 60px;
`;
const ControlBack = styled(ControlBackNext)`
  background-image: url(${carouselBack});
  margin-right: 15px;
`;
const ControlNext = styled(ControlBackNext)`
  background-image: url(${carouselNext});
  margin-left: 15px;
`;

export { ControlItem, ControlStyles, ControlBack, ControlNext };
