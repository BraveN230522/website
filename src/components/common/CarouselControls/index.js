import React from 'react';
import { Controller } from 'react-hook-form';

import { ControlStyles, ControlItem, ControlBack, ControlNext } from './style';

const CarouselControls = ({
  controlColor = 'controlColor',
  activeColor = 'white',
  items = [],
  activeIndex = 0,
  className,
  onItemClick,
  isMobile,
}) => {
  const backIndex = index => (index === 0 ? items.length - 1 : index - 1);
  const nextIndex = index => (index === items.length - 1 ? 0 : index + 1);
  return (
    <>
      {isMobile && <ControlBack onClick={onItemClick(backIndex(activeIndex))} />}
      <ControlStyles className={className} isMobile={isMobile}>
        {items.map((_, index) => (
          <ControlItem
            role={`control-${index + 1}`}
            activeColor={activeColor}
            controlColor={controlColor}
            key={index}
            onClick={onItemClick(index)}
            active={index === activeIndex}
          />
        ))}
      </ControlStyles>
      {isMobile && <ControlNext onClick={onItemClick(nextIndex(activeIndex))} />}
    </>
  );
};

export default CarouselControls;
