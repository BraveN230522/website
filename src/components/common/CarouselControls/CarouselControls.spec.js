import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import CarouselControls from './index';

const mockItems = [
  {
    image: 'some-image.jpg',
    title: 'New MCG Partnership',
    description:
      'Explore the inner sanctum of the MCG, walk on the hallowed arena and experience where legends play by taking a guided MCG tour.',
    link: '#',
  },
  {
    image: 'another-image.jpg',
    title: 'Item while we dont have graphql',
    description: 'Explore the inner sanctum of the MCG, walk on the hallowed arena.',
    link: '#',
  },
];

describe('<CarouselControls />', () => {
  describe('Snapshot Tests -', () => {
    it('Renders the CarouselControls without items', () => {
      const { container } = render(<CarouselControls />, {
        wrapper: withGlobals,
      });

      expect(container).toMatchSnapshot();
    });

    it('Renders the first control as selected when no activeIndex is passed', () => {
      const { container } = render(<CarouselControls items={mockItems} onItemClick={jest.fn()} />, {
        wrapper: withGlobals,
      });

      expect(container).toMatchSnapshot();
    });

    it('Renders a selected control item when an activeIndex is passed', () => {
      const { container } = render(<CarouselControls activeIndex={1} items={mockItems} onItemClick={jest.fn()} />, {
        wrapper: withGlobals,
      });

      expect(container).toMatchSnapshot();
    });
  });

  describe('Functional Tests -', () => {
    it('Calls onItemClick function passing the clicked control index as argument', () => {
      const onItemClickFn = jest.fn(() => () => null);
      const { getByRole } = render(<CarouselControls activeIndex={1} items={mockItems} onItemClick={onItemClickFn} />, {
        wrapper: withGlobals,
      });

      expect(onItemClickFn).toBeCalledWith(0);

      fireEvent.click(getByRole('control-2'));

      expect(onItemClickFn).toBeCalledWith(1);
    });
  });
});
