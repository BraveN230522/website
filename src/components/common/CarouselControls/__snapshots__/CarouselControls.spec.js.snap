// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselControls /> Snapshot Tests - Renders a selected control item when an activeIndex is passed 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  width: 31px;
  height: 5px;
  display: block;
  border-radius: 4px;
  background-color: #ffffff;
  margin-left: 4px;
  cursor: pointer;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.emotion-2:hover {
  width: 31px;
}

.emotion-0 {
  width: 13px;
  height: 5px;
  display: block;
  border-radius: 4px;
  background-color: #53A7FF;
  margin-left: 4px;
  cursor: pointer;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.emotion-0:hover {
  width: 31px;
}

<div>
  <div
    class="emotion-4 emotion-5"
  >
    <span
      class="emotion-0 emotion-1"
      role="control-1"
    />
    <span
      class="emotion-2 emotion-1"
      role="control-2"
    />
  </div>
</div>
`;

exports[`<CarouselControls /> Snapshot Tests - Renders the CarouselControls without items 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="emotion-0 emotion-1"
  />
</div>
`;

exports[`<CarouselControls /> Snapshot Tests - Renders the first control as selected when no activeIndex is passed 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  width: 31px;
  height: 5px;
  display: block;
  border-radius: 4px;
  background-color: #ffffff;
  margin-left: 4px;
  cursor: pointer;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.emotion-0:hover {
  width: 31px;
}

.emotion-2 {
  width: 13px;
  height: 5px;
  display: block;
  border-radius: 4px;
  background-color: #53A7FF;
  margin-left: 4px;
  cursor: pointer;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.emotion-2:hover {
  width: 31px;
}

<div>
  <div
    class="emotion-4 emotion-5"
  >
    <span
      class="emotion-0 emotion-1"
      role="control-1"
    />
    <span
      class="emotion-2 emotion-1"
      role="control-2"
    />
  </div>
</div>
`;
